### ISSUE TITLE
Consider replacing the websocket-client-simple gem with the same gem that is being used in the MiQ

### ISSUE THREAD

**Issue by jkremser** [hawkular/hawkular-client-ruby#201]
When we added this dependency, the MiQ didn't have the support for websocket communication, they add it with migration to rails 5 (iirc). Since the `websocket-client-simple` is no longer maintained and it has other issues like failing non-deterministically on segfaults (https://travis-ci.org/hawkular/hawkular-client-ruby/jobs/213159442#L438). Also there is currently no way to process incoming binary websocket frames (also failing with segfaults), so that we can't implement the JDR reports in the middleware provider.

**Comment by josejulio** [comment]
I agree those segfaults are a killer and really frequent. I have to restart tests almost always on PRs and even on master branch after merge.

If that lib is compatible with [ruby targets for this gem](https://github.com/hawkular/hawkular-client-ruby/blob/master/.travis.yml#L5-L6) I would say we should do it :+1:
